src/main/resources/twitter_config/getAccessToken.xml

```diff
@@ -0,0 +1,86 @@
+                <http method="post"
+                      uri-template="{uri.var.apiUrl}/2/oauth2/token?grant_type=refresh_token&amp;client_id={uri.var.clientId}&amp;refresh_token={uri.var.refreshToken}">
+                    <timeout>
+                        <duration>{$ctx:timeout}</duration>
+                        <responseAction>fault</responseAction>
+                    </timeout>
+                    <suspendOnFailure>
+                        <errorCodes>-1</errorCodes>
+                        <progressionFactor>1.0</progressionFactor>
+                    </suspendOnFailure>
+                    <markForSuspension>
+                        <errorCodes>-1</errorCodes>
+                    </markForSuspension>
+                </http>
+            </endpoint>
+        </call>
+        <property name="httpCode" expression="$axis2:HTTP_SC" scope="default" type="STRING"/>
+        <filter source="$ctx:httpCode" regex="20[0-9]">
+            <then>
+                <property name="uri.var.accessToken" expression="json-eval($.access_token)"/>
+                <property name="uri.var.refreshToken" expression="json-eval($.refresh_token)"/>
+
+                <!--         Setting the new access token to the registry. -->
+                <script language="js">
+                    <![CDATA[
+                    importPackage(Packages.org.apache.synapse.config);
+
+                    // Update accessToken value in registry
+                    var newAccessToken = mc.getProperty('uri.var.accessToken');
+                    var accessTokenRegistryPath = mc.getProperty('uri.var.accessTokenRegistryPath');
+                    var existingAccessToken =mc.getConfiguration().getRegistry().getResource(accessTokenRegistryPath);
+
+                    if(existingAccessToken == null){
+                    // Create the registry entry if no such entry exists.
+                    mc.getConfiguration().getRegistry().newResource(accessTokenRegistryPath,false);
+                    }
+                    // Update the registry entry.
+                    mc.getConfiguration().getRegistry().updateResource(accessTokenRegistryPath,newAccessToken);
+
+                    // Same flow for the refresh token
+                    var newRefreshToken = mc.getProperty('uri.var.refreshToken');
+                    var refreshTokenRegistryPath = mc.getProperty('uri.var.refreshTokenRegistryPath');
+                    var existingRefreshToken =mc.getConfiguration().getRegistry().getResource(refreshTokenRegistryPath);
+
+                    if(existingRefreshToken == null){
+                    mc.getConfiguration().getRegistry().newResource(refreshTokenRegistryPath,false);
+                    }
+                    mc.getConfiguration().getRegistry().updateResource(refreshTokenRegistryPath,newRefreshToken);
+                    ]]>
+                </script>
+
+                <property name="uri.var.accessToken.reg" expression="$ctx:uri.var.accessToken"/>
```

Comment:
Do we use this `uri.var.accessToken.reg` property in the rest of the mediation flow?

Reply:
By setting the registry value here, developer can make the next proxy server call without the access token value.